jacobian_follower/src/jacobian_follower/jacobian_follower.cpp

```diff
@@ -812,3 +813,46 @@ PointSequence JacobianFollower::get_tool_positions(std::vector<std::string> tool
   std::transform(tool_transforms.cbegin(), tool_transforms.cend(), std::back_inserter(positions), get_translation);
   return positions;
 }
+Eigen::Matrix4Xd JacobianFollower::getLinkToRobotTransform(std::vector<std::string> joint_names,
+                                                           std::vector<double> joint_positions, std::string link_name) {
+  robot_state::RobotState state(model_);
+  state.setVariablePositions(joint_names, joint_positions);
+  auto const &transform = state.getGlobalLinkTransform(link_name);
+  return transform.matrix();
+}
+
+std::vector<Eigen::Matrix4Xd> JacobianFollower::getLinkToRobotTransforms(
+    std::vector<std::string> const &joint_names, std::vector<double> const &joint_positions,
+    robot_state::RobotStatePtr state, std::vector<std::string> const &link_names) const {
+  state->setVariablePositions(joint_names, joint_positions);
+  std::vector<Eigen::Matrix4Xd> transforms;
+  auto get_transform = [&](std::string const &link_name) {
+    auto const &transform = state->getGlobalLinkTransform(link_name);
+    return transform.matrix();
+  };
+  std::transform(link_names.cbegin(), link_names.cend(), std::back_inserter(transforms), get_transform);
+  return transforms;
+}
+
+std::vector<Eigen::Matrix4Xd> JacobianFollower::getLinkToRobotTransforms(
+    std::vector<std::string> const &joint_names, std::vector<double> const &joint_positions,
+    std::vector<std::string> const &link_names) const {
+  auto state = std::make_shared<robot_state::RobotState>(model_);
+  return getLinkToRobotTransforms(joint_names, joint_positions, state, link_names);
+}
+
+std::vector<std::vector<Eigen::Matrix4Xd>> JacobianFollower::batchGetLinkToRobotTransforms(
+    std::vector<std::vector<std::string>> const &joint_names, std::vector<std::vector<double>> const &joint_positions,
+    std::vector<std::string> const &link_names) const {
```

Comment:
Add some input validation: 
```if joint_names.size() != joint_positions.size() {throw something}```

If you throw inside a pybind function, what does the user see? Hopefully some useful error message.

Reply:
https://pybind11.readthedocs.io/en/stable/advanced/exceptions.html?highlight=exception

even better, you can catch them as python exceptions!

Reply:
Amazing